src/services/addressProvider.ts

```diff
@@ -40,15 +48,44 @@ export class AddressProvider<T extends ChainId> extends Service {
       case 42161:
         return "0xcAd10033C86B0C1ED6bfcCAa2FF6779938558E9f";
       default:
-        throw new Error("Unsupported chain id");
+        throw new Error(`Unsupported chain id: ${chainId}`);
```

Comment:
This is not related to this PR, added this because I wrote the test for this function and I feel it's helpful when debugging to have the reason for the throw (in this case, the `chainId`)